src/System/reachability.rs

```diff
@@ -1,41 +1,40 @@
 use edbm::zones::OwnedFederation;
 
-use crate::component::LocationType;
+use super::query_failures::PathFailure;
+use super::specifics::SpecificPath;
 use crate::ModelObjects::component::{State, Transition};
-use crate::TransitionSystems::{LocationID, TransitionSystem};
+use crate::Simulation::decision::Decision;
+use crate::TransitionSystems::{LocationID, TransitionSystemPtr};
 use std::collections::{HashMap, VecDeque};
 use std::rc::Rc;
 
+use super::query_failures::PathResult;
+
 /// This holds the result of a reachability query
+#[derive(Debug, Clone)]
 pub struct Path {
-    pub path: Option<Vec<Transition>>,
-    pub was_reachable: bool,
+    pub path: Vec<Decision>,
 }
+
 // This holds which transition from which state (the destination_state of the previous_sub_path) we took to reach this state
 struct SubPath {
     previous_sub_path: Option<Rc<SubPath>>,
     destination_state: State,
-    transition: Option<Transition>,
+    transition: Option<(Transition, String)>,
 }
 
 fn is_trivially_unreachable(start_state: &State, end_state: &State) -> bool {
+    // If any of the zones are empty
+    if start_state.zone_ref().is_empty() || end_state.zone_ref().is_empty() {
+        return true;
+    }
+
     // If the end location has invariants and these do not have an intersection (overlap) with the zone of the end state of the query
     if let Some(invariants) = end_state.get_location().get_invariants() {
         if !end_state.zone_ref().has_intersection(invariants) {
             return true;
         }
     }
-    // If the start state is a universal or inconsistent location and the end state isn't
-    // Since universal and inconsistent locations cannot reach other locations
-    if matches!(
-        start_state.decorated_locations.loc_type,
-        LocationType::Universal | LocationType::Inconsistent
-    ) && !start_state
-        .decorated_locations
-        .compare_partial_locations(&end_state.decorated_locations)
-    {
-        return true;
-    }
 
     false
 }
```

Comment:
Can be simplified:
```suggestion
    // If any of the zones are empty
    if start_state.zone_ref().is_empty() || end_state.zone_ref().is_empty() {
        true
    }
    // If the end location has invariants and these do not have an intersection (overlap) with the zone of the end state of the query
    else if let Some(invariants) = end_state.get_location().get_invariants() {
        !end_state.zone_ref().has_intersection(invariants)
    } else {
        false
    }
```
Can even be simplified even further, but I don't think it would add anything
```rust
    // If any of the zones are empty
    start_state.zone_ref().is_empty() || end_state.zone_ref().is_empty() ||
            // If the end location has invariants and these do not have an intersection (overlap) with the zone of the end state of the query
            if let Some(invariants) = end_state.get_location().get_invariants() {
            !end_state.zone_ref().has_intersection(invariants)
        } else { false }
```

Reply:
I personally think the returns here are fine. The semantics are even different between your suggestion and the previous. Consider we want to add a new condition in the future:
```rust
    if start_state.zone_ref().is_empty() || end_state.zone_ref().is_empty() {
        true
    }
    else if let Some(invariants) = end_state.get_location().get_invariants() {
        !end_state.zone_ref().has_intersection(invariants)
    } else if new_cond() {
        true
    } else {
        false
    } 
```
Then `new_cond()` is never evaluated if the `end_state` has invariants, even if it has an intersection. You could of course put it before the invariant check, but are we sure that we will remember that we must do that in the future?

Reply:
That's a good point, lets just leave it as is